content/docs/iac/concepts/debugging/_index.md

```diff
@@ -20,288 +20,61 @@ Because Pulumi uses general purpose programming languages to provision cloud res
-
-```yaml
-runtime:
-  name: go
-  options:
-    binary: ./debug.sh
-```
-
-Run the `pulumi up` command:
-
-```bash
-$ pulumi up
-Previewing update (dev)
-
- Type                 Name      Plan       Info
- pulumi:pulumi:Stack  dev       create...  warning layer=rpc Listening for remote connections
-```
-
-Open your Pulumi program in your IDE and set a breakpoint at the line where you want the debugger to stop.
-
-Navigate to your IDE and connect the debugger to the same port as specified in the script (2345 in our case).
-
-For example, in [VS Code](https://code.visualstudio.com/docs/editor/debugging), you can define the following `.vscode/launch.json` file:
-
-```json
-{
-    "version": "0.2.0",
-    "configurations": [
-        {
-            "name": "Connect to server",
-            "type": "go",
-            "request": "attach",
-            "mode": "remote",
-            "remotePath": "${workspaceFolder}",
-            "port": 2345,
-            "host": "127.0.0.1"
-        }
-    ]
-}
-```
-
-Navigate to the `Run and Debug` menu on the left pane, and then click the Start Debugging (F5) button with the `Connect to server` configuration.
-
-Now you can step through the program and inspect variables. Once completed, resume the execution and let the Pulumi CLI complete.
-
-## Debugging Java programs
-
-This approach assumes your project uses Maven and you have Maven shell scripts on your path.
-
-Configure the runtime options in `Pulumi.yaml` to use `mvnDebug`:
-
-```yaml
-runtime:
-  name: java
-  options:
-    use-executor: "mvnDebug"
-```
-
-Open your Pulumi program in your IDE and set a breakpoint at the line where you want the debugger to stop.
-
-Navigate to your IDE and attach to the port 8000. For example, in [VS Code](https://code.visualstudio.com/docs/editor/debugging), you can define the following `.vscode/launch.json` file:
-
-```json
-{
-    "version": "0.2.0",
-    "configurations": [
-        {
-            "type": "java",
-            "name": "Debug (Attach)",
-            "projectName": "App",
-            "request": "attach",
-            "hostName": "localhost",
-            "port": 8000
-        }
-    ]
-}
-```
-
-Navigate to the `Run and Debug` menu on the left pane, and then click the Start Debugging (F5) button with the `Debug (Attach)` configuration.
-
-Now you can step through the program and inspect variables. Once completed, resume the execution and let the Pulumi CLI complete.
-
-## Debugging YAML programs
-
-It's not possible to debug a YAML program with a step-by-step debugger because YAML is a configuration file, not an executable program.
+1. Start with an existing Pulumi project that contains `Pulumi.yaml` and your program.
+2. Open your IDE and set a breakpoint in your program.
+3. Run `pulumi up --attach-debugger` from a command line.
+4. Wait until `pulumi` is paused waiting for a debugger to attach.
+5. Attach the debugger to a running process of your language runtime.
```

Comment:
Is it worth mentioning that the full debug configuration can be found in the event log, if the user passes `--event-log` (and enables that with PULUMI_DEBUG_COMMANDS=true)?  It's a bit more complicated though, so may not be worth it.